packages/calendar-time-utils/src/calendar-time.ts

```diff
@@ -162,3 +163,57 @@ export const parseInputText = (
   }
   return '';
 };
+
+const localizedDateFormatPatternCache = new Map<string, string>();
+export const getLocalizedDateTimeFormatPattern = (
+  locale: string,
+  formatType: 'date' | 'time' | 'full' = 'date'
+): string => {
+  const key = `${locale}::${formatType}`;
+  if (localizedDateFormatPatternCache.has(key)) {
+    return localizedDateFormatPatternCache.get(key)!;
+  }
+
+  // References:
+  //  https://momentjs.com/docs/#/i18n/locale-data/
+  //  https://momentjs.com/docs/#/displaying/ ("Localized formats" section)
+  const localeData = moment().locale(locale).localeData();
+  let localizedFormat = '';
+  switch (formatType) {
+    case 'date':
+      localizedFormat = localeData.longDateFormat('L');
+      break;
+    case 'time':
+      localizedFormat = localeData.longDateFormat('LT');
+      break;
+    case 'full':
+      localizedFormat = `${localeData.longDateFormat(
+        'L'
+      )} - ${localeData.longDateFormat('LT')}`;
+      break;
+  }
```

Comment:
I thought about it but it seemed to me it was more clear to have all the variants and no default since we know one of those would be used because we are typing it.
Thinking again about it, as users don't need to use Typescript, perhaps its better.

I can think in two options:
```
  let localizedFormat = '';
  switch (formatType) {
    case 'time':
      localizedFormat = localeData.longDateFormat('LT');
      break;
    case 'full':
      localizedFormat = `${localeData.longDateFormat(
        'L'
      )} - ${localeData.longDateFormat('LT')}`;
      break;
    case 'date': // <--- Not really needed, but it think having it here makes it more clear to the reader
    default:
      localizedFormat = localeData.longDateFormat('L');
      break;
  }
```

```
  // Default format is 'date'
  let localizedFormat = localeData.longDateFormat('L');
  switch (formatType) {
    case 'time':
      localizedFormat = localeData.longDateFormat('LT');
      break;
    case 'full':
      localizedFormat = `${localeData.longDateFormat(
        'L'
      )} - ${localeData.longDateFormat('LT')}`;
      break;
  }
```

Reply:
Ok so what we can do is throw an error in the `default` case, so that we make sure that we cover all known cases but throw an error otherwise.

Reply:
Updated in this commit: 6edef8f5850fc17e4134f815ad652e396aeafcb8